library/src/image.cxx

```diff
@@ -353,6 +353,24 @@ bool image::operator!=(const image& reference) const
   return !this->operator==(reference);
 }
 
+//----------------------------------------------------------------------------
+double image::operator()(int x, int y, int channel) const
+{
+  double v = this->Internals->Image->GetScalarComponentAsDouble(x, y, 0, channel);
+
+  switch (getChannelType())
+  {
+    case ChannelType::BYTE:
+      return v / 255.0;
+    case ChannelType::SHORT:
+      return v / 65535.0;
+    default:
+      break;
+  }
+
+  return v;
```

Comment:
```suggestion
  return this->Internals->Image->GetScalarComponentAsDouble(x, y, 0, channel);
```
Normalization feels specific to the palette usage, `image` API random access makes sense but should stay generic IMO

Reply:
I'm not convinced by the current approach either.

Reply:
Maybe it makes sense, but lets make sure it makes sense because the API makes sense, not because we need it to behave like that to support that feature, iykwim.

Reply:
In rendering, passing integers (could be `char` or `short`) is considered as "normalized integers", and sampling buffers (or textures) results in a value between 0 and 1.  
If you think about it, a 8-bit channel in an image is just defining the brightness of the channel, it's a *quantized* value of the (0,1) range.  
What I'm doing here by dividing by 255 or 65535 is a dequantization (I think I made up this word, but you get it)  
32-bits floating point images can go beyond the (0,1) range, and it's called HDR.

See:
- https://www.khronos.org/opengl/wiki/Normalized_Integer
- https://pytorch.org/docs/stable/quantization.html

Reply:
Looks like it is indeed a concept that exist and could be expected, lets make it clear in the doc what getContent and [] provides.

Reply:
> In rendering, [...]

That's a better argument than "byte as double is weird" ;p

to sum up:

1. if `f3d::image` is stated to represent a generic image, ie. where a pixel is one or more arbitrary numerical value, then the pixel access shouldn't make assumptions about the meaning of the data. It would still need to cast to `vector<double>` because that can hold any of the possible underlying `byte`/`short`/`float` values but that's it, any transformation is up to the user to know about and deal with. Examples where you wouldn't want the API to mess with your data: indexed images where the values are actually ids, images where the values are raw physical data (DEM), anything where the quantization does not map to `[0,1]` (signed difference? normal map?).

2. if `f3d::image` is stated to represent a "rendering image", ie. where a pixel represents one or more color intensity (or otherwise normalized quantity), then it's fair to say the underlying encoding is an implementation detail and the pixel access API operates on/with said normalized values, that does mean making assumptions and assigning meaning to the data but if that's explicitly the nature/purpose of the class then all good. Users wanting to deal with esoteric bitmaps can still do so through the data buffer and the pixel access API is just not for them.

I'm personally partial to number 1, but, given the scope of F3D, number 2 is definitely valid.

Reply:
But "bytes as double is weird" is still one of my arguments 😛 
C++ is a typed language, and even if `f3d::image` abstracts the underlying type, trying to get values as `double` when it's not stored as such internally is conceptually an issue to me.  
If we want random access of not normalized pixel, I'd prefer to have `getPixelAsByte`, `getPixelAsShort` and `getPixelAsFloat` that throws if the requested type is invalid.

Reply:
> abstracts the underlying type [...] throws if the requested type is invalid.

We don't have the same definition of abstraction then :D
Coming from python it feels to me like setting yourself such runtime traps on purpose in a strongly and statically typed (and compiled!) language is a "worst of both world" situation. Obviously having the fully templated `image<Type, Size, Count>` would quickly become a drag for obvious reasons but if you care that much about types wouldn't you at least have `image<Type>` and throw on load so at least your accesses are type-safe? (or maybe I'm _that_ misguided in my understanding of c++)

Anyway we're in purely rhetorical territory now (because who doesn't love arguing moot points with strangers on the internet) so maybe we'll need to move this discussion somewhere else

Reply:
> We don't have the same definition of abstraction then :D

Maybe I should have said `f3d::image` is generic instead? 🤔 

We cannot really have an `image<Type>` because you don't know at compile time the type of your image if you are loading it from file. We could throw at opening if it's not the correct type though but that's a different design.

Reply:
Just read your message again and realized I'm describing exactly what you suggested 🤦  
Anyway, I'm not against it, but that's a big change, it needs in-depth discussions definitely out of the scope of this PR.  
Also, I don't know what it would look like in Python? 3 different Python classes for each specialization (like `f3d.Image8`, `f3d.Image16` and `f3d.Image32`)?

Reply:
no worries; and I'm not even saying it should be changed at all, just basically rambling out my confusion :)

For python idk, it's not statically typed so you could do the "throw on wrong type" thing or you just return whatever type and let the client check what it gets at runtime. For the actual data storage you'd probably just use numpy arrays for efficiency/performance and compatibility with existing ecosystem?